### PR TITLE
Update resources and sensors in separate thread

### PR DESCRIPTION
**Pull request by kenchiang**
Reviewer: @jnealtowns 
Refactor onlp-snmpd platform and sensor handling so that updates happen in a separate thread.

### CODE REVIEW

File: packages/base/any/onlp-snmpd/builds/src/onlp_snmp/module/src/onlp_snmp_sensors.c [hunk 1]
```diff
@@ -918,59 +972,115 @@ collect_sensors__(onlp_oid_t oid, void* cookie)
...
-            ss->now_valid = false;
+            get_next_info(ss)->valid = false;
         }
     }
 
-    /* discover new sensors for all tables */
+    /* discover new sensors for all tables,
+     * writing validity into next_info for all sensors */
     onlp_oid_iterate(ONLP_OID_SYS, 0, collect_sensors__, NULL);
+    AIM_LOG_INFO("iteration complete");
```
Comment (jnealtowns) [inline packages/base/any/onlp-snmpd/builds/src/onlp_snmp/module/src/onlp_snmp_sensors.c]: Lets remove this log message.

File: packages/base/any/onlp-snmpd/builds/src/onlp_snmp/module/src/onlp_snmp_platform.c [hunk 2]
```diff
@@ -88,22 +91,47 @@ resource_int_register(int index, const char* desc,
...
+static void
+resource_update(void)
 {
     uint64_t now = aim_time_monotonic();
-    if (now - resource_update_time >
+    if (now - last_resource_update_time >
         (ONLP_SNMP_CONFIG_RESOURCE_UPDATE_SECONDS * 1000 * 1000)) {
-        resource_update_time = now;
+        last_resource_update_time = now;
         AIM_LOG_INFO("update resource objects");
```
Comment (jnealtowns) [inline packages/base/any/onlp-snmpd/builds/src/onlp_snmp/module/src/onlp_snmp_platform.c]: Lets remove this log message.

File: packages/base/any/onlp-snmpd/builds/src/onlp_snmp/module/src/onlp_snmp_sensors.c [hunk 3]
```diff
@@ -918,59 +972,115 @@ collect_sensors__(onlp_oid_t oid, void* cookie)
...
+        return;
+    }
+
+    if (restructure_trigger) {
+        AIM_LOG_INFO("restructure has not happened, skip sensor update");
+        return;
+    }
+
+    last_sensor_update_time = now;
+    AIM_LOG_INFO("update sensor objects");
```
Comment (jnealtowns) [inline packages/base/any/onlp-snmpd/builds/src/onlp_snmp/module/src/onlp_snmp_sensors.c]: Remove or move to TRACE

File: packages/base/any/onlp-snmpd/builds/src/onlp_snmp/module/src/onlp_snmp_sensors.c [hunk 4]
```diff
@@ -918,59 +972,115 @@ collect_sensors__(onlp_oid_t oid, void* cookie)
...
                 }
             }
         }
     }
 
+    /* swap front and back buffers */
+    swap_curr_next_info();
+
+    /* trigger table restructuring */
+    AIM_LOG_INFO("trigger restructure");
```
Comment (jnealtowns) [inline packages/base/any/onlp-snmpd/builds/src/onlp_snmp/module/src/onlp_snmp_sensors.c]: Remove or TRACE

**Comment by kenchiang** [comment]
pushed new commit addressing review comments; please take a look.  thanks.

**Pull request merged**